tests/rainbowGrades/test_csci1100.py

```diff
@@ -0,0 +1,295 @@
+
+    # Get the Submitty repo path
+    # script_path = os.path.dirname(os.path.realpath(__file__))
+    repository_path = "__INSTALL__FILLIN__SUBMITTY_REPOSITORY__"
+    script_path = os.path.join("__INSTALL__FILLIN__SUBMITTY_INSTALL_DIR__","test_suite","rainbowGrades")
+    runner_dir = os.path.join("__INSTALL__FILLIN__SUBMITTY_DATA_DIR__","to_be_graded_batch")
+
+    # Verify resources exist, set up initial temporary directories and configuration
+    print("Creating temporary RainbowGrades test directories")
+    test_tmp = tempfile.mkdtemp("", "")
+    print("Made new directory {}".format(test_tmp))
+
+    if not os.path.exists(test_tmp):
+        error_and_cleanup(test_tmp, "Failed to create temporary directory")
+
+    for f in os.listdir(runner_dir):
+        if "__csci1100__" in f:
+            error_and_cleanup(test_tmp,"csci1100 has assignments in the grading queue."
+                                       " Wait for the autograder to finish and then generate new grade summary reports"
+                                       "prior to re-running this test.")
+
+    rainbow_path = os.path.join(repository_path,"RainbowGrades")
+    if not os.path.exists(rainbow_path):
+        error_and_cleanup(test_tmp, "Couldn't find Rainbow Grades source code")
+
+    rainbow_tmp = os.path.join(test_tmp, "rainbow_grades")
+    os.mkdir(rainbow_tmp)
+
+    summary_tmp = os.path.join(test_tmp, "grade_summaries")
+    os.mkdir(summary_tmp)
+
+    grading_tmp = os.path.join(test_tmp, "grading")
+    os.mkdir(grading_tmp)
+
+    if not os.path.exists(rainbow_tmp) or not os.path.exists(summary_tmp):
+        error_and_cleanup(test_tmp, "Failed to create temporary subdirectory")
+
+    print("Copying Rainbow Grades code from Submitty to RainbowGrades")
+    try:
+        for f in os.listdir(rainbow_path):
+            shutil.copy(os.path.join(rainbow_path, f), rainbow_tmp)
+    except Exception as e:
+        error_and_cleanup(test_tmp, "{}".format(e))
+
+    # Copy non-standard files over
+    try:
+        shutil.copy(os.path.join(script_path, "MakefileHelperTest"), os.path.join(rainbow_tmp, "MakefileHelper"))
+        shutil.copy(os.path.join(script_path, "Makefile_csci1100"), os.path.join(summary_tmp, "Makefile"))
+        shutil.copy(os.path.join(script_path, "customization_csci1100.json"),
+                    os.path.join(summary_tmp, "customization.json"))
+        shutil.copy(os.path.join(repository_path, "grading", "json_syntax_checker.py"),
+                    os.path.join(grading_tmp, "json_syntax_checker.py"))
+    except Exception as e:
+        error_and_cleanup(test_tmp, "{}".format(e))
+
+    # Update Makefile to use the temporary location of RainbowGrades
+    try:
+        with open(os.path.join(summary_tmp, "Makefile"), 'r') as make_file:
+            make_file_contents = make_file.readlines()
+        with open(os.path.join(summary_tmp, "Makefile"), 'w') as make_file:
+            for line in make_file_contents:
```

Comment:
Ah, I missed that. Yeah, you could use `r+` and use `seek()` and such to navigate the file and change it inline while reading it, it's not a huge file to read into memory and this is simpler.